### PR TITLE
Increase test coverage (nc_generator and filesize)

### PR DESCRIPTION
**Pull request by cairosanders**
Addresses #6 
Adds tests for `nc_generator` and `netcdf_file.filesize`

### CODE REVIEW

File: pupynere.py [hunk 1]
```diff
@@ -145,6 +145,8 @@ def REVERSE(nptype):
                dtype(np.int32):   NC_INT,
                dtype(np.int64):   NC_INT,  # will be converted to int32
                dtype(np.float32): NC_FLOAT,
+               dtype('>f4'):      NC_FLOAT,
```
Comment (jameshiebert) [inline pupynere.py]: Are these not the same as `np.float32` and `np.float64`?

File: pupynere.py [hunk 2]
```diff
@@ -1135,14 +1137,14 @@ def byteorderer(target):
     except StopIteration:
         pass
 
-def check_byteorder(input):
+def check_byteorder(input_):
```
Comment (jameshiebert) [inline pupynere.py:1138]: Good call!

File: pupynere.py [hunk 3]
```diff
@@ -1135,14 +1137,14 @@ def byteorderer(target):
...
     ''' an iteration-based approach to byteorderer above
         input should be an iterable of numpy arrays with which to
         fill the netcdf file
     '''
     try:
         while True:
-            data = input.next()
+            data = input_.next()
```
Comment (jameshiebert) [inline pupynere.py]: In Python3, I think that one is supposed to use the [`next()` builtin function](https://docs.python.org/3/library/functions.html#next). Should we start doing that now?

File: tests/test_generators.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        # Test filesize property of a virtual file with nonrecvars
+        assert f.filesize == 88
+
+        # Test generator
+        pipeline = nc_generator(f, _input())
+        with tempfile.NamedTemporaryFile(suffix=".nc") as fn:
+            for block in pipeline:
+                fn.write(block)
+
+            nc = netcdf_file(fn.name, 'r')
```
Comment (jameshiebert) [inline tests/test_generators.py:36]: Should this be outside of the `with tempfile...` block to ensure that the output gets flushed to the file, before you try to open it?

File: tests/test_generators.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        with self.assertRaises(ValueError):
+            f.filesize
+
+        # Test generator
+        pipeline = nc_generator(f, _input())
+        with tempfile.NamedTemporaryFile(suffix=".nc") as fn:
+            for block in pipeline:
+                fn.write(block)
+
+            nc = netcdf_file(fn.name, 'r')
```
Comment (jameshiebert) [inline tests/test_generators.py:72]: Likewise.

File: tests/test_generators.py [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    def runTest(self):
+        f = netcdf_file(None, 'w')
+        temp = f.createVariable(VAR_NAME,VAR_TYPE)
+        temp.assignValue(VAR_VAL)
+
+        # Test filesize property of a virtual file with nonrecvars
+        assert f.filesize == 88
+
+        # Test generator
+        pipeline = nc_generator(f, _input())
```
Comment (jameshiebert) [inline tests/test_generators.py:30]: Cool, good test.

**Review by jameshiebert — COMMENTED** [review]
Looks pretty good to me, aside from a few small comments? By how much does this increase the test coverage?

**Review by corviday — APPROVED** [review]
Nice job! I checked [pydap.responses.netcdf](https://github.com/pacificclimate/pydap.responses.netcdf/blob/master/src/pydap/responses/netcdf/__init__.py) and it looks like these tests cover all the functionality used by that package, so we should be in a much stronger place to upgrade and make sure it behaves correctly going forward.

**Pull request merged**